cloudinit/config/cc_puppet.py

```diff
@@ -118,6 +118,14 @@ def __init__(
         self.csr_attributes_path = csr_attributes_path
 
 
+def get_service_name():
+    release = util.lsb_release()["codename"]
+    service = "puppet-agent"
+    if release in ("bionic", "focal", "jammy", "kinetic"):
+        service = "puppet"
+    return service
```

Comment:
other modules have a hash like this: https://github.com/canonical/cloud-init/blob/009377174f9a5224218f24c37036ff226dffa08d/cloudinit/config/cc_ntp.py#L50-L91

followed by distro specific overrides:

https://github.com/canonical/cloud-init/blob/009377174f9a5224218f24c37036ff226dffa08d/cloudinit/config/cc_ntp.py#L94-L216

this general approach might make some sense here: we could default to the new method, and allow a config selection to choose the old method

this would also allow us to encode different config paths, like FreeBSD for example uses.

Reply:
@igalic good suggestion on teasing out distribution config path defaults generally speaking. But, the case in the puppet module looks like it actually shells out to discover the configured defaults on disk, so the separate distro config file paths are unnecessary as the installed puppet (wherever you are) will tell us on the cmdline via a call to: [puppet config print <setting>](https://github.com/canonical/cloud-init/blob/main/cloudinit/config/cc_puppet.py#L149)

Reply:
Also, given that the binary in question is either puppet-agent (forward-looking) vs puppet for this module, I don't think we want to hardcode expected puppet* binary names per distro as pip installed vs packaged puppet binaries may change/evolve over time. So baking in the try <new_puppet>/except <old_puppet> seems to cover the primary/secondary use-cases without having to keep cloud-init implementation in step with each distro's update timeline for puppet package updates.